providers/json-binding/src/main/java/org/jboss/resteasy/plugins/providers/jsonb/AbstractJsonBindingProvider.java

```diff
@@ -46,7 +46,7 @@ public static Charset getCharset(final MediaType mediaType) {
          String charset = mediaType.getParameters().get("charset");
          if (charset != null) return Charset.forName(charset);
       }
-      return Charset.defaultCharset();
+      return Charset.forName("utf-8");
```

Comment:
Thinking about this fix again, it seems that this doesn't fully fix [RESTEASY-2171](https://issues.jboss.org/browse/RESTEASY-2171) (cc @asoldano @ronsigal ). Consider this end-point:
```java
    @GET
    @Produces("application/json;charset=us-ascii")
    @Path("a")
    public CustomObject a() throws Exception {
        return new CustomObject().setA("öü");
    }
```
And consider this RFC7159 part:

>  Note: No "charset" parameter is defined for this registration. Adding one really has no effect on compliant recipients.

Jackson2 returns UTF8 encoding, but JSON-B returns us-ascii encoding in this specific case.

Reply:
I understand then that Jackson2 is wrong. If you clearly state the charset then it should be used, as JSON-B is doing.

Reply:
See RFC7159: Adding one (charset) really has no effect on compliant recipients. @asoldano, @ronsigal WDYT about this?

Reply:
Honestly, I'm not super sure here, but I'd stick with the behavior that Jackson2 provider had so far (returning UTF8), which didn't lead to complaints afaiu (we're seeing the problem with JSON-B now). And the RFC7159 seems to confirm this.